### ISSUE TITLE
plot instantaneous firing rate instead of histogram counts?

### ISSUE THREAD

**Issue by jasmainak** [jonescompneurolab/hnn-core#448]
If we divide the histogram counts by the bin width, we can get instantaneous firing rate:

https://jonescompneurolab.github.io/hnn-core/stable/generated/hnn_core.viz.plot_spikes_hist.html#hnn_core.viz.plot_spikes_hist

To me it seems that would be more interpretable, e.g., if there is pathological firing etc. Wdyt @ntolley @rythorpe @cjayb ?

**Comment by ntolley** [comment]
This sounds great to me. In general I think the plotting functions could do with a some improvement with respect to interfacing with matplotlib.

What would be great to add is an optional `kwargs**` that are passed directly to the plot call. It may be tricky in terms of maintaining the look of certain plots, but I think this address issues in the past with respect to colors and axis limits.

**Comment by rythorpe** [comment]
> If we divide the histogram counts by the bin width, we can get instantaneous firing rate:
> 
> https://jonescompneurolab.github.io/hnn-core/stable/generated/hnn_core.viz.plot_spikes_hist.html#hnn_core.viz.plot_spikes_hist

I agree with this, however, we'll need to be thoughtful about how we calculate spike rates for drive cells, as the spike rate for an evoked all-to-all drive cell that targets all 270 real cells could either be calculated as 270 spikes/time_bin or 1 spikes/time_bin/target_cell. Conversely, the spike rate for a non-synchronous, cell-specific drive cell that connects 1-to1 with a single real cell can only be calculated as 1 spike/time_bin = 1 spike/time_bin/target_cell.

**Comment by jasmainak** [comment]
> I agree with this, however, we'll need to be thoughtful about how we calculate spike rates for drive cells, as the spike rate for an evoked all-to-all drive cell that targets all 270 real cells could either be calculated as 270 spikes/time_bin or 1 spikes/time_bin/target_cell. Conversely, the spike rate for a non-synchronous, cell-specific drive cell that connects 1-to1 with a single real cell can only be calculated as 1 spike/time_bin = 1 spike/time_bin/target_cell.

Isn't the firing rate independent of the cells it's connected to? So if we're talking about drive cells it would be 270 spikes / time bin / n_drive_cells. Now if it's non-synchronous, you'd have n_drive_cells = 270 and if it's all-to-all, then you'd have n_drive_cells = 1

**Comment by HimalayanSaswataBose** [comment]
Reopening this issue, found this to be an interesting track to get to know the codebase. I am trying to resolve this issue. Do I just change the histogram counts to get the instantaneous firing rate (and accordingly change the documentation)? or do I need to make few more changes for the edit to stay consistent with the system? I am new to this field, so I do not know much about the theoretical implications of the results and the code, but I hope I can make meaningful contributions to the codebase with the little knowledge and experience I have.
